src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_Int16.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class PrimitiveParser
+    {
+        public static partial class InvariantUtf16
+        {
+            public unsafe static bool TryParseInt16(char* text, int length, out short value)
+            {
+                int consumed;
+                var span = new ReadOnlySpan<byte>(text, length * 2);
+                return PrimitiveParser.TryParseInt16(span, out value, out consumed, EncodingData.InvariantUtf16);
+            }
+            public unsafe static bool TryParseInt16(char* text, int length, out short value, out int charactersConsumed)
+            {
+                var span = new ReadOnlySpan<byte>(text, length * 2);
+                int bytesConsumed;
+                bool result = PrimitiveParser.TryParseInt16(span, out value, out bytesConsumed, EncodingData.InvariantUtf16);
+                charactersConsumed = bytesConsumed / 2;
+                return result;
+            }
+            public static bool TryParseInt16(ReadOnlySpan<char> text, out short value)
+            {
+                int consumed;
+                var byteSpan = text.Cast<char, byte>();
+                return PrimitiveParser.TryParseInt16(byteSpan, out value, out consumed, EncodingData.InvariantUtf16);
+            }
+            public static bool TryParseInt16(ReadOnlySpan<char> text, out short value, out int charactersConsumed)
+            {
+                var byteSpan = text.Cast<char, byte>();
+                int bytesConsumed;
+                bool result = PrimitiveParser.TryParseInt16(byteSpan, out value, out bytesConsumed, EncodingData.InvariantUtf16);
+                charactersConsumed = bytesConsumed / 2;
+                return result;
+            }
+
+            public static partial class Hex
+            {
+                public unsafe static bool TryParseInt16(char* text, int length, out short value)
+                {
+                    int consumed;
+                    var span = new ReadOnlySpan<byte>(text, length * 2);
+                    return PrimitiveParser.TryParseInt16(span, out value, out consumed, EncodingData.InvariantUtf16, 'X');
```

Comment:
The parser can be made to parse 0xFF and 0xff (and perhaps 0xFf) as long as we think we can do this in a performant manner. I am unsure about the direction we should go if we determine that having "catch-all" Hex parsing APIs is not what we want.